aws/build.rake

```diff
@@ -166,45 +154,32 @@ if (rack_env?(:staging) && CDO.name == 'staging') || rack_env?(:development)
       RakeUtils.system 'rm', '-f', deploy_dir('rebuild-apps')
     end
   end
+else
+  APPS_COMMIT_TASK = build_task('apps-commit') {}
+end
 
-  #
-  # Define the CODE_STUDIO COMMIT task. If CODE_STUDIO_COMMIT_TASK produces new output, that output needs to be
-  #   committed because it's input for the DASHBOARD task.
-  #
-  CODE_STUDIO_COMMIT_TASK = build_task('code-studio-commit', [deploy_dir('rebuild'), CODE_STUDIO_TASK]) do
-    code_studio_changed = false
-    Dir.chdir(dashboard_dir('public/code-studio-package')) do
-      code_studio_git_status = `git status --porcelain .`
-      code_studio_changed = !code_studio_git_status.strip.empty?
-      HipChat.log "<b>code-studio</b> package changes:\n#{code_studio_git_status}" if code_studio_changed
-    end
+#
+# Define the CODE STUDIO BUILD task.
+#
+CODE_STUDIO_NODE_MODULES = Dir.glob(code_studio_dir('node_modules', '**/*'))
+CODE_STUDIO_BUILD_PRODUCTS = [code_studio_dir('npm-debug.log')] + Dir.glob(code_studio_dir('build', '**/*'))
+CODE_STUDIO_SOURCE_FILES = Dir.glob(code_studio_dir('**/*')) - CODE_STUDIO_NODE_MODULES - CODE_STUDIO_BUILD_PRODUCTS
```

Comment:
You could probably make the argument that it's not.

Right now, the code inside this block will not even get run if all that changed was node_modules (for example). In a world where we remove this list, we would check for a commit_hash file, compare the value there to git_latest_commit_hash, and if all that changed was node_modules, but there were no new commits, we would be done.

So yeah, I think we can remove these and make this simpler.